examples/point-cloud-laz/app.js

```diff
@@ -0,0 +1,177 @@
+
+  for (let i = 0; i < points.length; i++) {
+    points[i].position[0] = (points[i].position[0] - xMid) / scale;
+    points[i].position[1] = (points[i].position[1] - yMid) / scale;
+    points[i].position[2] = (points[i].position[2] - zMid) / scale;
+  }
+}
+
+class Example extends PureComponent {
+
+  constructor(props) {
+    super(props);
+
+    this._onChangeViewport = this._onChangeViewport.bind(this);
+    this._onInitialized = this._onInitialized.bind(this);
+    this._onResize = this._onResize.bind(this);
+    this._onUpdate = this._onUpdate.bind(this);
+
+    this.state = {
+      width: 0,
+      height: 0,
+      points: [],
+      progress: 0,
+      viewport: {
+        lookAt: [0, 0, 0],
+        distance: 10,
+        rotationX: 0,
+        rotationY: 0,
+        fov: 30,
+        minDistance: 0.1,
+        maxDistance: 10
+      }
+    };
+  }
+
+  componentWillMount() {
+    window.addEventListener('resize', this._onResize);
+    this._onResize();
+  }
+
+  componentDidMount() {
+    this._canvas.fitBounds([-0.5, -0.5, -0.5], [0.5, 0.5, 0.5]);
+
+    const {points} = this.state;
+
+    const skip = 10;
+    loadLazFile('data/indoor.laz').then(rawData => {
+      parseLazData(rawData, skip, (decoder, progress) => {
+        for (let i = 0; i < decoder.pointsCount; i++) {
+          const {color, position} = decoder.getPoint(i);
+          points.push({color, position});
+        }
+
+        if (progress >= 1) {
+          normalize(points);
+        }
+
+        this.setState({points, progress});
+      });
+    });
+
+    window.requestAnimationFrame(this._onUpdate);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this._onResize);
+  }
+
+  _onResize() {
+    const {innerWidth: width, innerHeight: height} = window;
+    this.setState({width, height});
+  }
+
+  _onInitialized(gl) {
+    gl.clearColor(0.07, 0.14, 0.19, 1);
+    gl.enable(gl.DEPTH_TEST);
+    gl.depthFunc(gl.LEQUAL);
+  }
+
+  _onChangeViewport(viewport) {
+    this.setState({viewport: {...this.state.viewport, ...viewport}});
+  }
+
+  _onUpdate() {
+    const {viewport} = this.state;
+    this.setState({
+      viewport: {
+        ...viewport,
+        rotationY: viewport.rotationY + 1
+      }
+    });
+    window.requestAnimationFrame(this._onUpdate);
+  }
+
+  _renderLazPointCloudLayer() {
+    return this.state.points.length && new PointCloudLayer({
+      id: 'laz-point-cloud-layer',
+      data: this.state.points,
+      projectionMode: COORDINATE_SYSTEM.IDENTITY,
+      getPosition: d => d.position,
+      getNormal: d => [0, 0.5, 0.2],
+      getColor: d => [255, 255, 255, 128],
+      radiusPixels: 1
+    });
+  }
+
+  _renderDeckGLCanvas() {
+    const {width, height, viewport} = this.state;
+    const canvasProps = {width, height, ...viewport};
+    const glViewport = OrbitController.getViewport(canvasProps);
+
+    return width && height && <OrbitController {...canvasProps} ref={canvas => {
```

Comment:
Nit: the line break formatting here isn't very pretty. I'd use a parenthesis around the component, put it on its own line, and then one prop per line.